[16.0][FIX] mail_environment: add search field for fetchmail.server's is_ssl

StefanRijnhart opened the pull request:
Fixes

```
while parsing /home/odoo/16.0/addons/mail/views/fetchmail_views.xml:88
Error while validating view near:

                    <separator/>
                    <filter string="SSL" name="ssl" domain="[('is_ssl', '=', True)]" help="If SSL required."/>
                    <separator/>
                    <filter string="Archived" name="inactive" domain="[('active', '=', False)]"/>
                </search>
Unsearchable field 'is_ssl' in path 'is_ssl' in domain of <filter name="ssl"> ([('is_ssl', '=', True)]))
```

While upgrading --all on a setup with mail_environment installed.

florian-dacosta commented:
thanks for the fix
There is small pre-commit issue.
About the test, I am no expert, but IMHO just a small test doing a search with the `is_ssl` field in the domain and checking the result is correct would be more than enough.

StefanRijnhart committed:
[FIX] mail_environment: add search field for fetchmail.server's is_ssl

Fixes

```
while parsing /home/odoo/16.0/addons/mail/views/fetchmail_views.xml:88
Error while validating view near:

                    <separator/>
                    <filter string="SSL" name="ssl" domain="[('is_ssl', '=', True)]" help="If SSL required."/>
                    <separator/>
                    <filter string="Archived" name="inactive" domain="[('active', '=', False)]"/>
                </search>
Unsearchable field 'is_ssl' in path 'is_ssl' in domain of <filter name="ssl"> ([('is_ssl', '=', True)]))
```

While upgrading --all on a setup with mail_environment installed.

StefanRijnhart commented:
@florian-dacosta Thanks! Indeed, we just want to test the search method but I had to figure out how to add a server environment configuration test (and how any of this works in the first place). Now proud to add the first tests in this module!

florian-dacosta reviewed (approved):
@StefanRijnhart I think a simple "write" works to fill the value, it should work as a default value for the field if there are no server environment config.
But with a config it is fine too (maybe better so it also tests that adding a server env config works!) thanks again and LGTM

rven commented:
Could someone merge this?

florian-dacosta commented:
@simahawk @etobella possible to merge this one?

simahawk commented:
/ocabot merge patch

StefanRijnhart commented:
Pre-commit fixed in https://github.com/OCA/server-env/pull/148

bguillot commented:
/ocabot merge patch
